index.d.ts

```diff
@@ -0,0 +1,54 @@
+/// <reference types="json-rpc-protocol" />
+/// <reference types="node" />
+
+import { EventEmitter } from "events";
+import { JsonRpcPayload, JsonRpcParamsSchema } from "json-rpc-protocol";
+
+export as namespace JsonRpcPeer;
+
+export * from "json-rpc-protocol";
+
+export class Peer extends EventEmitter implements NodeJS.WritableStream {
+  constructor(onmessage?: (message: JsonRpcPayload, data: any) => Promise<any>);
+
+  public exec(
+    message: string | object,
+    data?: any
```

Comment:
I checked the code again, and it seems that we have never used the `data` argument?

Could you please have a look at it to clarify the usage of the `data`?

Reply:
It is used, it passed to `_handle`/`onMessage`.

Reply:
Ok, I now know it's useful.

It seems there has nothing about it in the documents, we can improve it later.